src/renderers/dom/fiber/ReactDOMFiberEntry.js

```diff
@@ -757,8 +758,27 @@ function createPortal(
   return ReactPortal.createPortal(children, container, null, key);
 }
 
+type WorkNode = {
+  commit(): void,
+
+  _reactRootContainer: *,
+  _expirationTime: ExpirationTime,
+};
+
+function Work(root: *, expirationTime: ExpirationTime) {
+  this._reactRootContainer = root;
+  this._expirationTime = expirationTime;
+}
+Work.prototype.commit = function() {
+  const root = this._reactRootContainer;
+  const expirationTime = this._expirationTime;
+  DOMRenderer.unblockRoot(root, expirationTime);
+  DOMRenderer.flushRoot(root, expirationTime);
+};
+
 type ReactRootNode = {
   render(children: ReactNodeList, callback: ?() => mixed): void,
```

Comment:
I wonder if this new API really should return a Thenable since that's what people have been asking for with these callback APIs. I noticed because the prerender API doesn't have it.